includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/WPForms_Event_List.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\WPForms_Event_List
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events;
+
+/**
+ * Class for containing tracking event information for WPForms plugin.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class WPForms_Event_List extends Measurement_Event_List {
+
+	/**
+	 * WPForms_Event_List constructor.
+	 *
+	 * @since n.e.x.t.
+	 */
+	public function __construct() {
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'WPForms',
+				'category'   => 'engagement',
+				'action'     => 'form_submit',
+				'selector'   => '.wpforms-submit-container button',
+				'on'         => 'click',
+				'metadata'   => <<<CALLBACK
+function( params, element ) {
+	var formId = element.closest('.wpforms-submit-container').querySelector('input[name="wpforms[id]"]').value;
```

Comment:
Flagging something here that we should do throughout all of these `metadata` callbacks: Let's code defensively. While chaining functions like this makes the code shorter, we should include checks to ensure these DOM functions return a result. For example, if `closest` returns `null`, right now it would throw a JavaScript error. And while that right now will probably never be the case, we don't control what these other plugins do so we need to put checks pretty much anywhere - it's a good practice either way.

Let's make this change throughout. If something returns a false-y value, stop proceeding and just return `params`. If anything, we could do a `console.error` in such a case, but if the DOM structure of a plugin changes, it must not result in an actual JS error.